packages/openapi3/src/openapi.ts

```diff
@@ -117,16 +120,53 @@ export function resolveOptions(
     ),
   };
 }
-// NOTE: These functions aren't meant to be used directly as decorators but as a
-// helper functions for other decorators.  The security information given here
-// will be inserted into the `security` and `securityDefinitions` sections of
-// the emitted OpenAPI document.
 
 export interface ResolvedOpenAPI3EmitterOptions {
   outputFile: string;
   newLine: NewLine;
 }
 
+/**
+ * Represents a node that will hold a JSON reference. The value is computed
+ * at the end so that we can defer decisions about the name that is
+ * referenced.
+ */
+class Ref {
```

Comment:
If we have to do the same thing for `cadl-autorest`, should some of these types be in the shared `openapi` library?

Reply:
Yes, I can do that. I had a lot more things in here that I moved to rest, but didn't move the remaining things to openapi. If it's OK with you, I'd prefer to do that separately.

Reply:
Sounds good!

Reply:
I will file an issue for this

Reply:
#1070